src/Processors/Transforms/WindowTransform.cpp

```diff
@@ -2388,6 +2398,16 @@ struct WindowFunctionLagLeadInFrame final : public WindowFunction
             return;
         }
 
+        if (argument_types.size() > 3)
+        {
+            throw Exception(ErrorCodes::TOO_MANY_ARGUMENTS_FOR_FUNCTION,
+                "Function '{}' accepts at most 3 arguments, {} given",
+                name, argument_types.size());
+        }
+
+        if (argument_types[0]->equals(*argument_types[2]))
+            return;
+
         const auto supertype = getLeastSupertype(DataTypes{argument_types[0], argument_types[2]});
```

Comment:
Not related to the changes, but `getLeastSupertype` throws an error itself (by default) so we miss a better error message with function name .

Reply:
Another issue:
```
SELECT lagInFrame(CAST(number, 'UInt64'), 2, NULL) OVER w
FROM numbers(50)
WINDOW w AS (PARTITION BY number % 20 ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING)

Code: 36. DB::Exception: The supertype 'Nullable(UInt64)' for the argument type 'UInt64' and the default value type 'Nullable(Nothing)' is not the same as the argument type. (BAD_ARGUMENTS) (version 24.8.1.1) (from 0.0.0.0:0) (in query: SELECT lagInFrame(number::UInt64, 2, NULL) OVER w FROM numbers(50) WINDOW w AS (PARTITION BY number % 20 rows between unbounded preceding and unbounded following);), Stack trace (when copying this message, always include the lines below):
```

I would expect this to work since the docs say `Value to return if calculated row exceeds the boundaries of the window frame. (Optional - null by default).`, so I was expecting NULL to just be ignored. Also not introduced here.

Reply:
Regarding getLeastSupertype - agree, will fix it.
Regarding default null - I would rather argue that doc is wrong - by default it's a default value for the column value type, so exception is correct, and doc requires a fix.